File: inventario_frontend/src/models/feeRule.ts
import { Product } from './product';

export class FeeRuleForCreation {
    public productId: string | undefined = undefined;
    public date: Date | undefined = undefined;
    public feesAmountTo: number | undefined = undefined;
    public percentage: number | undefined = undefined;
}

export class FeeRule extends FeeRuleForCreation {
    public id!: string;
    public product: Product | undefined = undefined;
}

File: Inventario/DataAccess/Models/FeeRule.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Data
{
    public class FeeRule: Entity
    {
        public Guid ProductId { get; set; }
        public Product Product { get; set; }
        public DateTime Date { get; set; }
        public int FeesAmountTo { get; set; }
        public decimal Percentage { get; set; }
    }
}
